falcon/app.py

```diff
@@ -886,3 +887,16 @@ def _get_body(self, resp, wsgi_file_wrapper=None):
             return iterable, None
 
         return [], 0
+
+
+# NOTE(mikeyusko): This function is temporary and will be deleted once
+# the API alias will be removed
+def api(*args, **kwargs):
```

Comment:
Sorry for the late notice as I realized this just now...

Could it potentially be a more robust approach to subclass `App` instead?
* The biggest difference is that the change would be less breaking for people doing something along the lines of `isinstance(something, API)`
* Following our discussion on Gitter, it would be a by all means legit class from the Sphinx perspective, allowing old docs to link to its decsription
* Your code emitting the warning can be easily ported to its constructor instead